jawa/constants.py

```diff
@@ -152,16 +152,16 @@ class MethodHandle(Constant):
 
     def __init__(self, pool, index, reference_kind, reference_index):
         super(MethodHandle, self).__init__(pool, index)
-        self._reference_kind = reference_kind
-        self._reference_index = reference_index
+        self.reference_kind = reference_kind
+        self.reference_index = reference_index
 
     @property
     def kind(self):
-        return self.pool.get(self._reference_kind)
+        return self.pool.get(self.reference_kind)
```

Comment:
The kind is just an enum, not a reference to another constant, per [JVMS §4.4.8](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-4.html#jvms-4.4.8):

> The value of the reference_kind item must be in the range 1 to 9. The value denotes the kind of this method handle, which characterizes its bytecode behavior ([§5.4.3.5](https://docs.oracle.com/javase/specs/jvms/se8/html/jvms-5.html#jvms-5.4.3.5)).

This property thus probably isn't needed, and just having the `reference_kind` field works.